### PR TITLE
Clarify that Broker caching for groupBy v2 queries does not work

### PR DESCRIPTION
**Pull request by FrankChen021**
### Description

Broker caching has been disabled for groupBy v2 queries since #3950, and this limitation is stated  under the section [Differences between v1 and v2](https://druid.apache.org/docs/latest/querying/groupbyquery.html#differences-between-v1-and-v2) of groupBy query doc. But since that's a large block of text, it's not intuitive for users to notice this constraint.

This PR adds a note under the section `Broker Caching` of the configuration doc to give users a more clear hint when they follow that doc to enable broker caching.

This PR has:
- [X] been self-reviewed.

### CODE REVIEW

File: docs/configuration/index.md [hunk 1]
```diff
@@ -1773,6 +1773,9 @@ You can optionally only configure caching to be enabled on the Broker by setting
 
 See [cache configuration](#cache-configuration) for how to configure cache settings.
 
+> Note: Even if cache is enabled, Brokers never cache group-by queries when [groupBy v2 strategy](../querying/groupbyquery.md#strategies), which is the default, is configured.
```
Comment (sthetland) [inline docs/configuration/index.md:1776]:
```suggestion
> Note: Even if cache is enabled, Brokers never cache groupBy queries when [groupBy v2 strategy](../querying/groupbyquery.md#strategies), which is the default, is configured.
```

**Review by sthetland — CHANGES_REQUESTED** [review]
Good idea to make this more prominent. Just one suggestion to put the `groupBy` term into its usual form.

**Comment by capistrant** [comment]
is groupBy v2 only not cacheable for the non-result cache or is it also not cacheable for result cache? If it is only for non-result cache, I think we should explicitly call that out to not confuse people.

**Comment by FrankChen021** [comment]
@capistrant Both of per-segment caching and whole-query caching are not supported for groupBy v2 queries. It's a good idea to put this in the doc to make it more clear.

**Review by capistrant — APPROVED** [review]
LGTM. the trick will be for someone to remember to remove this from the docs if down the road some day someone implements broker caching for groupby v2

**Comment by FrankChen021** [comment]
Hi @sthetland , Do you have any other comments ?

**Review by sthetland — APPROVED** [review]

**Pull request merged**